### PR TITLE
[chore] Delete language about vendor-specific components and obligated sponsorship

### PR DESCRIPTION
**Pull request by djaglowski**
I propose that we move to a purely voluntary sponsorship requirement going forward.

### CODE REVIEW

**Review by codeboten — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/new_component.yaml [hunk 1]
```diff
@@ -21,28 +21,18 @@ body:
...
+        Having a code owner is strongly recommended. However, you may use the issue to try to find a code owner for your component.
   - type: input
     attributes:
       label: Sponsor (optional)
       description: "A sponsor is an approver who will be in charge of being the official reviewer of the code.
-        For vendor-specific components, it's good to have a volunteer sponsor. If you can't find one, we'll assign one in a round-robin fashion.
-        For non-vendor components, having a sponsor means that your use-case has been validated.
-        If there are no sponsors yet for the component, it's fine: use the issue as a means to try to find a sponsor for your component."
+        Having a sponsor means that your use-case has been validated. If there are no sponsors yet for the component,
+        it's fine: use the issue as a means to try to find a sponsor for your component."
```
Comment (evan-bradley) [inline .github/ISSUE_TEMPLATE/new_component.yaml:35]:
```suggestion
        it's fine: use the issue as a means to try to find a sponsor for your component. If you are unable to find a sponsor, the component can still live in your own source repository and can be included in Collector distributions using the Collector builder."
```

**Review by evan-bradley — APPROVED** [review]
I think this makes sense at this stage of the project.

I included a suggestion for some phrasing that having a component in contrib isn't essential to having a component the issue opener or their users can leverage for their use cases. I think the fact that so many components are contained in contrib and components living in other repositories aren't widely publicized may lead to an impression that being in contrib is a requirement for having a usable component.

**Review by songy23 — APPROVED** [review]
I support

**Review by mwear — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]
As OpenTelemetry grows we're seeing more and more component proposals and are experiencing more new components that then get abandoned by their sponsors, leaving the community to pick up the slack. As we are stabilizing more and more collector APIs it becomes more feasible for component authors to maintain components in their own repositories. I agree that we have out-grown this rule.

**Pull request merged**